### PR TITLE
Fix flight1parse processing exception

### PR DESCRIPTION
**Pull request by minlp**
#### Description
Fix flight1parse processing exception
#### Reference issue
Fixes #...
https://github.com/pion/webrtc/issues/2629

### CODE REVIEW

**Review by adriancable — APPROVED** [review]
This all looks reasonable to me.

**Comment by adriancable** [comment]
This looks good to me, although I would like another pair of eyes on it as a sanity check before merging. @cnderrauber / @davidzhao / @Sean-Der ?

File: handshake_cache.go
Comment (davidzhao) [inline handshake_cache.go:125]: should out be nil in case it's not found through the loop?

**Review by davidzhao — COMMENTED** [review]
lgtm, just one question

**Review by cnderrauber — APPROVED** [review]
Looks good!

**Comment by adriancable** [comment]
@minlp - can you take a look at @davidzhao's comment and push a change to your fork if needed? Then I think we can get this merged.

**Comment by minlp** [comment]
> @minlp - can you take a look at @davidzhao's comment and push a change to your fork if needed? Then I think we can get this merged.

I have already updated the code on the fix_dtls_flight1parse_except branch.

**Comment by adriancable** [comment]
@minlp - ok great. Please squash the commit (you can retain the original title, “Fix flight1parse processing exception) and then assuming all checks are passed I will merge. Thank you!

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**
